File: Lesson04/IntentFilters/app/src/main/java/com/codingblocks/intentfilters/MainActivity.java
package com.codingblocks.intentfilters;

import android.content.Intent;
import android.net.Uri;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText etUrl;
    Button btnGo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etUrl = findViewById(R.id.etUrl);
        btnGo = findViewById(R.id.btnGo);

        btnGo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

//                Intent  intent = new Intent();
//                intent.setAction(Intent.ACTION_VIEW);
//                intent.setData(null);

                Intent i = new Intent(getBaseContext(), BrowserActivity.class);

//                Log.e("TAG", "onClick: Action is : " + i.getAction());

                String url = etUrl.getText().toString();
                i.setData(Uri.parse(url));

//                i.putExtra("EXPLICIT_INTENT_URL", url);
                startActivity(i);

            }
        });


//        Intent intent = getIntent();
//
//        Uri url = intent.getData();

        //Navigate the user to this above url

    }
}

File: Lesson18/Permissions/app/src/main/java/com/codingblocks/permissions/MainActivity.kt
package com.codingblocks.permissions

import android.Manifest
import android.content.Intent
import android.content.Intent.ACTION_CALL
import android.content.pm.PackageManager
import android.net.Uri
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.app.ActivityCompat
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    lateinit var callIntent: Intent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnCall.setOnClickListener {
            val input = "tel:${etNumber.text}"
            callIntent = Intent(ACTION_CALL).apply {
                data = Uri.parse(input)
            }

            if (
                ActivityCompat.checkSelfPermission(
                    this,
                    Manifest.permission.CALL_PHONE
                )
                == PackageManager.PERMISSION_GRANTED
            ) {
                startActivity(callIntent)
                //If the permission is granted
//                Toast.makeText(this, "Thank you for granting the permissions", Toast.LENGTH_SHORT).show()
            } else {
                //else request permissions
                ActivityCompat.requestPermissions(
                    this,
                    arrayOf(Manifest.permission.CALL_PHONE, Manifest.permission.SEND_SMS),
                    12345
                )
            }
        }
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<String>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)

        if (requestCode == 12345) {
            //This is the callback for phone permission

            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                //I have the calling permission
                startActivity(callIntent)

            } else {
                //Permission has been denied by the user, show him/her a toast
                Toast.makeText(this, "Sorry, I will need the calling permission to make calls", Toast.LENGTH_SHORT).show()
            }

        }

    }

}
